article/background.tex

```diff
@@ -4,107 +4,69 @@ \section{Background}
+Verification research has become more common due in part to dramatic results \cite{TODO: Salmon}, dedicated communities \cite{TODO: something wholesome}, and widely publicized large-scale reproducibility meta-analyses \cite{RPP}.
+Unfortunately, in Neuroimaging, the execution of data analysis code often remains non-trivial and time consuming. \cite{TODO: I made this up} 
+The execution of data analysis often requires significant knowledge of their software dependencies, which should be considered out-of-scope for the expected requirements of a researcher to reexecute.
+Because the installation, configuration, and execution software environments frequently requires in-depth knowledge of the internal workings, reexecution results remain underrepresented in scientific literature \cite{TODO my opinion, is that ok?}
+
+\subsection{Complexity in Reexecution}
+
+Reexecution is an irreplacible component of research reproduction, but it frequently requires significant time and uncommon expertise to attempt.
+What makes reexecution particularly challenging is that it requires integration of many components, and even small differences in software environments can cause the entire system to fail.
+Therefore, the authors consider all code that is executed TODO:(not neccesarily OS,etc tone it down) to be part of the requisite analysis instructions.
+This includes code that is installed as dependencies, which may come from many repositories that cannot be relied upon to host all legacy versions.
+Reexecution either requires updating the software to modern dependencies, or the creation of a similar computational environment.
+Environment construction becomes harder to reproduce with time as upstream repositories update, as the experienced with this project.
+
+\subsection{Software Dependency Problem and Containers}
+
+A fundamental challenge of software dependency management is that dependencies can be nested arbitrarily deeply, which increases in complexity (TODO: O^n? does that even make sense to say?)
+As soon as one version of one dependency eventually becomes unavailable, the first step of executability is manually solving a computationally difficult problem.
+One way of solving the issue of rebuilding environments with integrated dependencies is distributing an entire operating system image with the environment installed and configured, which can be emulated as a "guest" environment within the "host" machine.
+Virtual machines (colloquially "VMs") are attractive solutions to the software dependency problem, because the artifact that is distributed is in working order and should be executable with almost no knowledge of the internal workings.
+Once running, guest machines are self-contained and isolated from the host, which then eliminates the posibility of polluting the host environment and allows system administrators to safely allow relatively unrestricted usage from semi-trusted users.
```

Comment:
Package managers also do this. I think this is a too specific way of describing what you do, when, etc.... In the immediate term we use containers because people can't be bothered to use package managers.

Reply:
It is possible to use package managers for the same purpose, but I don't think there's much reason to discuss how that is done here.